keps/sig-instrumentation/1668-log-tracking/README.md

```diff
@@ -0,0 +1,372 @@
+
+All of three id's inception is from object creation and it dies with object deletion
+
+## Design Details
+
+### Prerequisite
+
+We need to consider three cases:
+- Case1: Requests from kubectl that creating an object
+- Case2: Requests from kubectl other than creating (e.g. updating, deleting) an object
+- Case3: Requests from controllers
+
+The design below is based on the above three cases
+
+The following picture show our design![design](./overview.png)
+
+
+
+we don't have any modifications in kubectl in this design.
+
+### Design of ID propagation in apiserver
+
+**1. Do othttp's original [Extract](https://pkg.go.dev/go.opentelemetry.io/otel/api/propagators#TraceContext.Extract)() to get [SpanContext](https://pkg.go.dev/go.opentelemetry.io/otel/api/trace#SpanContext)**
+
+- For request from kubectl, SpanContext is null,  do [Start](https://github.com/open-telemetry/opentelemetry-go/blob/3a9f5fe15f50a35ad8da5c5396a9ed3bbb82360c/sdk/trace/tracer.go#L38)() to start new SpanContext (new traceid and spanid)
+- For request from controller we can get a valid SpanContext(including traceid and spanid), do [Start](https://github.com/open-telemetry/opentelemetry-go/blob/3a9f5fe15f50a35ad8da5c5396a9ed3bbb82360c/sdk/trace/tracer.go#L38)() to update the SpanContext (new spanid)
+
+**2. Chain SpanContext and initialtraceid to "r.ctx"**
+
+- we use r.ctx to propagate those IDs to next handler
+
+**3. Make up a new outgoing [request](#design-of-id-progagation-in-client-go)**
+
+### Design of ID propagation (controller)
+
+**1. Extract SpanContext and initialtraceid from annotation of object to golang ctx**
+
+**2. Propagate golang ctx from objects to API Calls**
+
+When controllers create/update/delete an object A based on another B, we propagate context from B to A. E.g.:
+```
+    ctx = httptrace.SpanContextFromAnnotations(objB.GetAnnotations())
+    err = r.KubeClient.CoreV1().Create(ctx, objA...)
+```
+We do propagation across objects without adding traces to that components.
+
+**3. Make up a new outgoing [request](#design-of-id-progagation-in-client-go)**
+
+### Design of ID propagation in [client-go](https://github.com/kubernetes/client-go)
+client-go  helps to inject [TraceContext](https://pkg.go.dev/go.opentelemetry.io/otel/propagators#example-TraceContext) and [Baggage](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/baggage/api.md) to the outgoing http request header, something changes are like below:
+```diff
+@@ -868,6 +871,7 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
+                }
+                req = req.WithContext(ctx)
+                req.Header = r.headers
++               propagator := otel.NewCompositeTextMapPropagator(propagators.TraceContext{}, propagators.Baggage{})
```

Comment:
We should do this by using the [`WithPropagators()`](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/master/instrumentation/net/http/otelhttp/config.go#L104) option when using othttp in client-go.